src/System.Linq/tests/CountTests.cs

```diff
@@ -128,5 +118,34 @@ public void MultipleElementsAllMatchPredicate()
 
             Assert.Equal(expected, data.Count(IsEven));
         }
+
+        [Theory, MemberData("CountsAndTallies")]
+        public void CountMatchesTally<T, TEn>(T unusedArgumentToForceTypeInference, int count, TEn enumerable)
+            where TEn : IEnumerable<T>
+        {
+            Assert.Equal(count, enumerable.Count());
+        }
+
+        private static IEnumerable<object[]> EnumerateCollectionTypesAndCounts<T>(int count, IEnumerable<T> enumerable)
+        {
+            yield return new object[] { default(T), count, enumerable };
+            yield return new object[] { default(T), count, enumerable.ToArray() };
+            yield return new object[] { default(T), count, enumerable.ToList() };
+            yield return new object[] { default(T), count, new Stack<T>(enumerable) };
+        }
+
+        public static IEnumerable<object[]> CountsAndTallies()
+        {
+            int count = 100;
```

Comment:
Yes, it anything it would make it cheaper to add further variants of there was ever a reason to do so.